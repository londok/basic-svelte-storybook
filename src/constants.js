export const type = {
  primary: "primary",
  secondary: "secondary",
  transparent: "transparent",
};