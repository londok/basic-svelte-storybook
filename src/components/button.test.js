import { render, fireEvent, waitFor } from '@testing-library/svelte';
import { type } from "../constants"
import Button from './button.svelte';
const testId = "button-tid";
const buttonData = {
  title: null,
  type: null,
  click: null,
  disabled: false,
  loading: false,
}
describe('Button', () => {
  it('default', async () => {
    expect.hasAssertions();
const mockClick = jest.fn();
    buttonData.click = mockClick;
    const { getByTestId } = render(Button, { props: { data: buttonData } });
    let button;
    await waitFor(() => {
      button = getByTestId(testId);
      expect(button).toHaveTextContent('')
      expect(button.classList.contains(type.primary)).toBe(true);
      expect(button).toMatchSnapshot();
      fireEvent.click(button);
    })
    expect(mockClick).toHaveBeenCalledTimes(1);
  });
it('undefined', async () => {
    expect.hasAssertions();
const mockClick = jest.fn();
    buttonData.click = mockClick;
    const { getByTestId } = render(Button);
    await waitFor(() => {
      const button = getByTestId(testId);
      expect(button).toHaveTextContent('')
      expect(button.classList.contains("button")).toBe(true);
      expect(button).toMatchSnapshot();
      fireEvent.click(button);
    })
    expect(mockClick).toHaveBeenCalledTimes(0);
  });
it('title', async () => {
    expect.hasAssertions();
const { getByTestId } = render(Button, { props: { data: { ...buttonData, title: "Button Title" } } });
    await waitFor(() => {
      const button = getByTestId(testId);
      expect(button).toHaveTextContent("Button Title");
      expect(button).toMatchSnapshot();
    })
  });
it('loading', async () => {
    expect.hasAssertions();
const mockClick = jest.fn();
    buttonData.click = mockClick;
    const { getByTestId } = render(Button, { props: { data: { ...buttonData, loading: true } } });
    let button;
    await waitFor(() => {
      button = getByTestId(testId);
      expect(button).toHaveTextContent("Loading...")
      expect(button).toMatchSnapshot();
      fireEvent.click(button);
    })
    expect(mockClick).toHaveBeenCalledTimes(0);
  });
it('disabled', async () => {
    expect.hasAssertions();
const mockClick = jest.fn();
    buttonData.click = mockClick;
    const { getByTestId } = render(Button, { props: { data: { ...buttonData, disabled: true } } });
    let button
    await waitFor(() => {
      button = getByTestId(testId);
      expect(button).toHaveAttribute('disabled')
      expect(button).toMatchSnapshot();
      fireEvent.click(button);
    })
    expect(mockClick).toHaveBeenCalledTimes(0);
  })
  
  it('primary', async () => {
    expect.hasAssertions();
const { getByTestId } = render(Button, { props: { data: { ...buttonData, type: type.primary } } });
    await waitFor(() => {
      const button = getByTestId(testId)
      expect(button.classList.contains(type.primary)).toBe(true);
      expect(button).toMatchSnapshot();
    })
  })
it('secondary', async () => {
    expect.hasAssertions();
const { getByTestId } = await render(Button, { props: { data: { ...buttonData, type: type.secondary } } });
    await waitFor(() => {
      const button = getByTestId(testId)
      expect(button.classList.contains(type.secondary)).toBe(true);
      expect(button).toMatchSnapshot();
    })
  })
it('transparent', async () => {
    expect.hasAssertions();
const { getByTestId } = await render(Button, { props: { data: { ...buttonData, type: type.transparent } } });
    await waitFor(() => {
      const button = getByTestId(testId)
      expect(button.classList.contains(type.transparent)).toBe(true);
      expect(button).toMatchSnapshot();
    })
  })
});