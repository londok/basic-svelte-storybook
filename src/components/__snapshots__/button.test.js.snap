// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button default 1`] = `
<button
  class="button primary svelte-15uobuy"
  data-testid="button-tid"
>
  
</button>
`;

exports[`Button disabled 1`] = `
<button
  class=" svelte-15uobuy"
  data-testid="button-tid"
  disabled=""
>
  undefined
</button>
`;

exports[`Button loading 1`] = `
<button
  class=" svelte-15uobuy"
  data-testid="button-tid"
>
  Loading...
</button>
`;

exports[`Button primary 1`] = `
<button
  class="button primary svelte-15uobuy"
  data-testid="button-tid"
>
  
</button>
`;

exports[`Button secondary 1`] = `
<button
  class="button secondary svelte-15uobuy"
  data-testid="button-tid"
>
  
</button>
`;

exports[`Button title 1`] = `
<button
  class=" svelte-15uobuy"
  data-testid="button-tid"
>
  Button Title
</button>
`;

exports[`Button transparent 1`] = `
<button
  class="button transparent svelte-15uobuy"
  data-testid="button-tid"
>
  
</button>
`;

exports[`Button undefined 1`] = `
<button
  class="button primary svelte-15uobuy"
  data-testid="button-tid"
>
  
</button>
`;
